packages/grafana-data/src/field/fieldState.ts

```diff
@@ -159,33 +157,45 @@ export function calculateFieldDisplayName(field: Field, frame?: DataFrame, allFr
   return displayName;
 }
 
+let _frame: DataFrame | undefined;
```

Comment:
this will always leak one cached frame, but it is likely in-use elsewhere anyhow, and this cache is short-lived, since `getUniqueFieldName()` is called frequently

Reply:
how much more performant is this cache version of code vs just looping through frame fields - IMO that version was cleaner / easier to understand and maintain than current iteration 🤷‍♂️ (i.e. is this optimization necessary?)

Reply:
it saves a few hundred milliseconds in datasets with a few thousand fields, which is significant.

the new version simply has a early exit and cache. the only change in the function's logic is this extra block at the top -- it's not intermixed with the previous logic.

```js
    if (_frame !== frame) {
      _frame = frame;
      _allUnique = fields.length === new Set(fields.map(f => f.name)).size;
    }
```